apps/frontend/src/components/UserCreator.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState } from "react"
+import { useInvalidate } from "../libs/useApi";
+
+const formEncodeUser = (name: string, email: string) => {
+  return [
+    `name=${encodeURIComponent(name)}`,
+    `email=${encodeURIComponent(email)}`,
+  ].join('&')
+}
+
+const UserCreator = () => {
+  const invalidate = useInvalidate()
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  
+  const [result, setResult] = useState('');
+
+  const updateField = (type: 'name' | 'email') => (e: { target: { value: string; }; }) => {
+    if (type === 'name') {
+      setName(e.target.value)
+    } else {
+      setEmail(e.target.value)
+    }
+  }
+
+  const submit = () => {
+    // We shouldn't trust that the 'disabled' state on the button will be respected
+    // However this does mean duplicating our validation logic
+    if (!name || !email) return setResultText('Please enter both a name & email')
+
+    fetch('http://localhost:3001/api/1/users', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      body: formEncodeUser(name, email)
+    }).then(response => {
```

Comment:
I thought I got this working, but I was getting bad gateway errors at the end, with more time I would have investigated this further :)